codec/codectest/codectest.go

```diff
@@ -1153,3 +1156,26 @@ func FuzzStructUnmarshal(codec codecpkg.GeneralCodec, f *testing.F) {
 		require.Len(bytes, size)
 	})
 }
+
+func TestImplementsUnmarshalFrom(t testing.TB, codec codecpkg.GeneralCodec) {
+	require := require.New(t)
+
+	p := wrappers.Packer{MaxSize: 1024}
+	p.PackFixedBytes([]byte{0, 1, 2}) // pack 3 extra bytes prefix
+
+	mySlice := []bool{true, false, true, true}
+
+	require.NoError(codec.MarshalInto(mySlice, &p))
+
+	p.PackFixedBytes([]byte{7, 7, 7}) // pack 3 extra bytes suffix
+
+	bytesLen, err := codec.Size(mySlice)
+	require.NoError(err)
+	require.Equal(3+bytesLen+3, p.Offset)
+
+	p = wrappers.Packer{Bytes: p.Bytes, MaxSize: p.MaxSize, Offset: 3}
+
+	var sliceUnmarshaled []bool
+	require.NoError(codec.UnmarshalFrom(&p, &sliceUnmarshaled))
+	require.Equal(mySlice, sliceUnmarshaled)
```

Comment:
Can we test the state of `p` after calling `UnmarshalFrom`?
```suggestion
	var sliceUnmarshaled []bool
	require.NoError(codec.UnmarshalFrom(&p, &sliceUnmarshaled))
	require.Equal(mySlice, sliceUnmarshaled)
	require.Equal(
		wrappers.Packer{
			Bytes:   p.Bytes,
			MaxSize: p.MaxSize,
			Offset:  11,
		},
		p,
	)
```
or something

Reply:
after this lgtm